auto_hpl/build_run_hpl.sh

```diff
@@ -376,235 +376,234 @@ gpgcheck=0
+	eval "mkdir -p $HPL_PATH"
+	if [[ $? -ne 0 ]]; then
+		exit_out "\nUnable to create Directory $HPL_PATH. Try with sudo \n" 1
+	fi
+	cd $HPL_PATH
+	wget $HPL_LINK
+	if [ $? -ne 0 ]; then
+		exit_out "Error: wget $HPL_LINK failed." 1
+	fi
+	tar -xf hpl-$HPL_VER.tar.gz
+	if [ $? -ne 0 ]; then
+		exit_out "Error: tar -xf hpl-$HPL_VER.tar.gz failed" 1
+	fi
+	cd hpl-$HPL_VER
+
+	makefile=Make.Linux_${blaslib}
+	if [ $ubuntu -eq 1 ]; then
+		makefile=Make.Linux_${blaslib}_ubuntu
+	fi
+	if [ $aws -eq 1 ]; then
+		makefile=Make.Linux_${blaslib}_aws
+	fi
+	echo "sed s,TOPDIR,$run_dir, ${run_dir}/${makefile} > Make.Linux_${blaslib}"
+	sed s,TOPDIR,$run_dir, ${run_dir}/${makefile} > Make.Linux_${blaslib}
+	bindir=Linux_${blaslib}
+	make arch=Linux_${blaslib} 2>&1 > ${RESULTSDIR}/hpl_make.out
+	if [ $? -ne 0 ]; then
+		exit_out "Error: make arch=Linux_${blaslib} 2>&1 > ${RESULTSDIR}/hpl_make.out" 1
+	fi
 }
 
 clean_env()
 {
-  rm -rf $SCRIPT_DIR/blis
-  rm -rf $HPL_PATH
-  rm -rf $AMD_BLIS_DIR
+	rm -rf $SCRIPT_DIR/blis
+	rm -rf $HPL_PATH
+	rm -rf $AMD_BLIS_DIR
 }
 
 run_hpl()
 {
-  cd $HPL_PATH/hpl-$HPL_VER/bin/${bindir}
+	cd $HPL_PATH/hpl-$HPL_VER/bin/${bindir}
+
+	# ckup the existing HPL.dat file
+	mv HPL.dat HPL.dat.bkup
+	# Copy the HPL.dat file
+	cp $SCRIPT_DIR/HPL.dat .
+
+	num_mpi=$NUM_MPI_PROCESS_ST
+	outfile=${RESULTSDIR}/hpl-${blaslib}-$(date "+%Y.%m.%d-%H.%M.%S").log
+	if [[ $BLAS_MT -eq 1 ]]; then
+		if [[ -d /sys/devices/system/cpu/cpu0/cache/index3 ]]; then
+			bind_settings="--map-by l3cache"
+		else
+			bind_settings="--map-by socket"
+		fi
+		bind_settings="${bind_settings} -x OMP_NUM_THREADS=${NOMP}"
+		num_mpi=$NUM_MPI_PROCESS_MT
+	else
+		bind_settings="--bind-to core"
+	fi
+	echo "bind_settings=$bind_settings"
 
-  #ckup the existing HPL.dat file
-  mv HPL.dat HPL.dat.bkup
-  # Copy the HPL.dat file
-  cp $SCRIPT_DIR/HPL.dat .
-  
-  num_mpi=$NUM_MPI_PROCESS_ST
-  outfile=${RESULTSDIR}/hpl-${blaslib}-$(date "+%Y.%m.%d-%H.%M.%S").log
-  if [[ $BLAS_MT -eq 1 ]]; then
-    if [[ -d /sys/devices/system/cpu/cpu0/cache/index3 ]]; then
-      bind_settings="--map-by l3cache"
-    else
-      bind_settings="--map-by socket"
-    fi
-    bind_settings="${bind_settings} -x OMP_NUM_THREADS=${NOMP}"
-   num_mpi=$NUM_MPI_PROCESS_MT
-  else
-   bind_settings="--bind-to core"
-  fi
-  echo "bind_settings=$bind_settings"
-
-  echo  "$MPI_PATH/bin/mpirun --allow-run-as-root -np $num_mpi --mca btl self,vader --report-bindings $bind_settings ./xhpl"
-
-  echo "     T/V           N    NB     P     Q               Time                 Gflops"  > $outfile
-  for i in $(seq "$NUM_ITER")
-  do
-    $MPI_PATH/bin/mpirun --allow-run-as-root -np $num_mpi --mca btl self,vader --report-bindings $bind_settings ./xhpl 2>&1 > hpl.out
-    cat hpl.out | grep -E "WC|WR"  >> $outfile
-  done
-  cp $outfile $SCRIPT_DIR
-  cd $SCRIPT_DIR
+	echo  "$MPI_PATH/bin/mpirun --allow-run-as-root -np $num_mpi --mca btl self,vader --report-bindings $bind_settings ./xhpl"
```

Comment:
While we're editing, can we add echoing this to $outfile so the mpirun command line is stored in the same place as the result rather than having to go up a directory and munge through the full log.